### ISSUE TITLE
Odd phrasing makes C library dependent on C++

### ISSUE THREAD

**Issue by potswa** [cplusplus/draft#470]
[res.on.headers] §17.6.5.2/3 uses "shall" where the intent seems to be "may." It suggests that `<math.h>` must include `<cmath>`.

**Comment by jwakely** [comment]
No, the intent is to say the C headers don't include anything else, i.e. `<stdlib.h>` doesn't include `<vector>`. Changing that to "may" would weaken that.

This doesn't seem editorial anyway.

**Comment by potswa** [comment]
The intent is clear to me, but not to at least one [StackOverflow user](http://stackoverflow.com/questions/29454488/does-algorithm-include-cmath#comment47085396_29454488). It says "shall include only," but the intent is not to require that anything be included, which "shall" certainly does. The word "may" gives permission, which is the intent.

Good phrases here would be "may only include" or "shall not include any C++ header besides the corresponding one." If it's really not editorial, I can file a DR, but this is a much easier way to deal with a very small erratum.

**Comment by jwakely** [comment]
Maybe that wording should have been adjusted by http://cplusplus.github.io/LWG/lwg-defects.html#456

**Comment by potswa** [comment]
Yes, the current phrasing does make sense within the model described as a problem, but not with the resolution. What's the next step?

**Comment by jensmaurer** [comment]
Suggestion: "The C standard library headers (D.4) shall not include C++ standard library headers other than their corresponding C ++ standard library header, as described in 17.6.1.2."

**Comment by tkoeppe** [comment]
Note also that we have exhaustive synopses of the *cname* headers now, so we do kind of tell you what we expect to be in those headers.

**Comment by jensmaurer** [comment]
The issue is still that we need to prohibit C headers from including C++ headers. We don't want <stddef.h> to make std::vector available, ever.

**Comment by zygoloid** [comment]
How can we prevent that, given that <cstddef> is permitted to include <vector>? Seems like an LWG issue to me.

**Comment by jwakely** [comment]
Agreed, I think this should be dealt with as a DR.

**Comment by AlisdairM** [comment]
I see several proposals for good wording here --- should we open that LWG issue?

**Comment by jensmaurer** [comment]
Sure, feel free to do so.
